### PR TITLE
[CI] Add GitHub Actions CI

### PR DESCRIPTION
**Pull request by anayini**
## Changes in this pull request

Add GitHub Actions configuration for running
CI.  Also remove the travis configuration.  I think
using GitHub Actions will be a little nicer since its
a bit better integrated and Travis seems to be a lot
slower.

### Checklist

- [x] All tests pass. Demo project builds and runs.
- [x] I added tests, an experiment, or detailed why my change isn't tested.
- [x] I added an entry to the `CHANGELOG.md` for any breaking changes, enhancements, or bug fixes.
- [x] I have reviewed the [contributing guide](https://github.com/plangrid/ReactiveLists/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+      - dev
+  pull_request:
+    branches:
+      - master
```
Comment (ronaldsmartin) [inline .github/workflows/ci.yml]: Wouldn't we want to run tests on all PR branches, not just the mainline ones?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  iOS:
+    name: iOS Test
+    runs-on: macOS-latest
+    strategy:
+      matrix:
+        destination: ["OS=13.2,name=iPhone 11"]
+    steps:
+      - uses: actions/checkout@v1
+      - name: Switch to Xcode 11
+        run: sudo xcode-select --switch /Applications/Xcode_11.1.app
```
Comment (ronaldsmartin) [inline .github/workflows/ci.yml]: Is this step necessary if we've specified a specific `IOS_SDK` above?

File: .travis.yml [hunk 3]
```diff
@@ -1,50 +0,0 @@
...
-
-script:
-  - set -o pipefail
-  - xcodebuild -version -sdk
-
-  - if [ $POD_LINT == "YES" ]; then
-        bundle exec pod lib lint;
-    fi
-
-  - xcodebuild build -workspace "$WORKSPACE" -scheme "$EXAMPLE_SCHEME" -sdk "$SDK" -destination "$DEST" ONLY_ACTIVE_ARCH=NO | bundle exec xcpretty -c;
```
Comment (ronaldsmartin) [inline .travis.yml]: We are missing an equivalent step that makes sure the Example app continues to build. Can we add that too?

File: .travis.yml [hunk 4]
```diff
@@ -1,50 +0,0 @@
...
-
-  - xcodebuild test -workspace "$WORKSPACE" -scheme "$FRAMEWORK_SCHEME" -sdk "$SDK" -destination "$DEST" -configuration Debug -enableCodeCoverage YES ONLY_ACTIVE_ARCH=NO | bundle exec xcpretty -c
-
-  - bundle exec danger
```
Comment (ronaldsmartin) [inline .travis.yml]: And Danger

File: .travis.yml [hunk 5]
```diff
@@ -1,50 +0,0 @@
...
-
-after_success:
-  - bash <(curl -s https://codecov.io/bash)
```
Comment (ronaldsmartin) [inline .travis.yml]: And code coverage? That `enableCodeCoverage YES` flag isn't quite equivalent to this, is it?

File: .travis.yml (hunk 3 above)
Comment (anayini) [reply .travis.yml]: Oh yeah def

File: .travis.yml (hunk 5 above)
Comment (anayini) [reply .travis.yml]: Yeah I think I still need to do this - I was searching to see if there was already an action - but yeah I'll figure out how to add this

File: .github/workflows/ci.yml (hunk 2 above)
Comment (anayini) [reply .github/workflows/ci.yml]: Oh true don't need this because of `DEVELOPER_DIR`

File: .github/workflows/ci.yml (hunk 1 above)
Comment (anayini) [reply .github/workflows/ci.yml]: Hm yeah maybe just all `pull_request` and `push` events?

**Comment by anayini** [comment]
@ronaldsmartin - Gonna come back and do Danger later if thats okay with you?  I need to regenerate a new `DANGER_GITHUB_API_TOKEN` and I still have to figure out how we've been doing that in the past (do we use a special user or something?).  After I generate the token though, Github now has supports for repository level secret stores, so it should be easy.

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+      - name: Bundle Install
+        run: bundle install
+      - name: Install Pods
+        run: bundle exec pod install
+      - name: Run Tests
+        run: |
+          set -o pipefail
+          xcodebuild clean
+          xcodebuild test -workspace "$WORKSPACE" -scheme "$LIBRARY_SCHEME" -destination "${{ matrix.destination }}" -configuration Debug -enableCodeCoverage YES ONLY_ACTIVE_ARCH=NO | bundle exec xcpretty -c
+          xcodebuild test -workspace "$WORKSPACE" -scheme "$EXAMPLE_APP_SCHEME" -destination "${{ matrix.destination }}" -configuration Debug -enableCodeCoverage YES ONLY_ACTIVE_ARCH=NO | bundle exec xcpretty -c
```
Comment (ronaldsmartin) [inline .github/workflows/ci.yml]: Non-blocking, but let's move this into a separate step in the future?

**Comment by ronaldsmartin** [comment]
Also you should show this during Platform Meeting because it's cool.

**Pull request merged**